libsolidity/interface/CompilerStack.cpp

```diff
@@ -539,11 +552,15 @@ void CompilerStack::link()
 
 vector<string> CompilerStack::contractNames() const
 {
-	if (m_stackState < AnalysisPerformed)
-		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Parsing was not successful."));
 	vector<string> contractNames;
-	for (auto const& contract: m_contracts)
-		contractNames.push_back(contract.first);
+
```

Comment:
I wonder if I maybe should make this return value based on the imports instead?

Reply:
I don't understand what you are suggesting here, but do you think it makes sense to just keep the code here and instead initialize `m_contracts` already right after parsing?

Reply:
I removed this change completely now. Turns out it was useless.